From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:36:46 -0800
Subject: [PATCH] wifi-scripts: add lpi_psd and lpi_bcn_enhance hostapd config
 options

Support configuration of LPI PSD and beacon enhance mode.

Declare lpi_psd and lpi_bcn_enhance in the wifi-device schema and emit
them from the device config generator in usr/share/ucode/wifi/hostapd.uc
when enabled.

Fixes: SDG-1871

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/schema/wireless.wifi-device.json
+++ b/src/usr/share/schema/wireless.wifi-device.json
@@ -515,6 +515,16 @@
 			"minimum": 0,
 			"maximum": 4
 		},
+		"lpi_bcn_enhance": {
+			"description": "Enable 6GHz LPI beacon enhancement mode",
+			"type": "boolean",
+			"default": false
+		},
+		"lpi_psd": {
+			"description": "Enable 6GHz LPI power spectral density mode",
+			"type": "boolean",
+			"default": false
+		},
 		"macaddr": {
 			"type": "alias",
 			"default": "bssid"
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -543,6 +543,11 @@ function generate(config) {
 	if (config.acs_exclude_6ghz_non_psc)
 		append('acs_exclude_6ghz_non_psc', 1);
 
+	if (config.lpi_psd)
+		append('lpi_psd', 1);
+	if (config.lpi_bcn_enhance)
+		append('lpi_bcn_enhance', 1);
+
 	/* TX Power */
 	append_vars(config, [ 'min_tx_power' ]);
 
